packages/adapter-squid/src/services/EvmChainService.ts

```diff
@@ -0,0 +1,72 @@
+import * as chains from "viem/chains";
+import { PublicClient, createPublicClient, http } from "viem";
+import { singleton } from "tsyringe";
+
+import { IEvmChainService } from ".";
+import { erc20Abi } from "../constants";
+
+@singleton()
+export class EvmChainService implements IEvmChainService {
+   private publicClients: Map<number, PublicClient>;
+   private initialised: boolean = false;
+
+   constructor() {
+      this.publicClients = new Map<number, PublicClient>();
+   }
+
+   public async init(clientConfigs: Map<number, string>): Promise<void> {
+      if (this.initialised) {
+         throw new Error("EvmChainProvider is already initialised");
+      }
+
+      for (const [chainId, rpcUrl] of clientConfigs.entries()) {
+         const thisChain = this.getChainById(chainId);
+         const thisTransport = rpcUrl ? http(rpcUrl) : http();
+
+         const client = createPublicClient({
+            chain: thisChain,
+            transport: thisTransport,
+         });
+         this.publicClients.set(chainId, client);
+      }
+   }
+
+   public async getBalance(
+      chainId: number,
+      address: string,
+      tokenAddress: string
```

Comment:
I think viem exports an address type we can use for these two (and not have to do the `as `0x{string}`)